python/siti.py

```diff
@@ -70,7 +68,14 @@ def analyze_video(video):
         "ti": TiFeatures()
     }
     i = 0
-    for frame in skvideo.io.vreader(video):
+    for frame in skvideo.io.vreader(video, as_grey=True):
+        if len(frame.shape) > 2:
+            width = frame.shape[-2]
+            height = frame.shape[-3]
+            frame = frame.reshape((height, width))
+
+        frame = frame.astype('float32')
```

Comment:
Could you maybe explain why this is needed? Just for curiosity. As there are no decimal values in the color range, an 8 bit integer should have been enough, or at least that was the understanding when implementing it.

Reply:
Yes, it is true, but to calculate the TI one frame is subtracted by another and the new array can have negative numbers.  To calculate the SI the Sobel filter returns an array of the same input type with values larger than 255. Then I change that to float32. In the opencv code this is also done in the [line 328 fom main.cpp code](https://github.com/Telecommunication-Telemedia-Assessment/SITI/blob/ca6e804934ee4842a4dcfd2ed17c783142fe0624/src/SITI/main.cpp#L328).